src/sqlfluff/core/parser/segments/raw.py

```diff
@@ -148,3 +151,51 @@ def get_end_pos_marker(self):
     def get_start_pos_marker(self):
         """Return the pos marker at the start of this segment."""
         return self.pos_marker
+
+
+class RawSegmentFactory:
+    optional = False
+    _anti_template = None
+
+    def __init__(self, cls, classname, _template, _name, **kwargs):
+        # if classname.startswith("semicolon"):
+        #     import pdb; pdb.set_trace()
+        self.cls = cls
+        self.__name__ = classname
+        self._template = _template
+        self._name = _name
+        self._kwargs = kwargs
+
+    def __repr__(self):
+        return f"<{self.cls.__name__}: {self.__name__} {self._template} {self._name} {self._kwargs}"
```

Comment:
I think the `repr` should be more explicit that this is a `RawSegmentFactory` rather than trying to disguise it as the other class. Unless I'm missing something?

Reply:
Probably so. This was to help me with debugging the parser -- I'm flipping back and forth between this code and the `master` branch, so it's been helpful thus far if these objects resemble the traditional ones. Can definitely revisit this once things are working better.